tests/resource/test_resource.py

```diff
@@ -147,3 +151,88 @@ def test_ipython_display(self, r, rep, capsys):
         r._ipython_display_()  # pylint: disable=protected-access
         captured = capsys.readouterr()
         assert rep in captured.out
+
+
+def _construct_tape_from_ops(lst_ops):
+    with QuantumTape() as tape:
+        for op in lst_ops:
+            qml.apply(op)
+    return tape
+
+
+class _CustomOpWithResource(Operation):  # pylint: disable=too-few-public-methods
+    num_wires = 2
+    name = "CustomOp1"
+
+    # TODO: add test where custom op has depth >1 once Circuit Graph PR is merged.
+    def resources(self):
+        return Resources(
```

Comment:
We expect the users to create and return a Resources object in their custom operations instead of resource information? I think it makes sense.

Reply:
Yes, this decision comes directly from the PRD. I think it packages everything nicely, but I can change it if we prefer something else?